lib/features/modeling/util/ModelingUtil.js

```diff
@@ -37,19 +42,42 @@ export function getParent(element, anyType) {
  * Determines if the local modeling direction is vertical or horizontal.
  *
  * @param {Element} element
+ * @param {ElementRegistry} elementRegistry
  *
  * @return {boolean} false for vertical pools, lanes and their children. true otherwise
  */
-export function isDirectionHorizontal(element) {
+export function isDirectionHorizontal(element, elementRegistry) {
 
-  var types = [ 'bpmn:Participant', 'bpmn:Lane' ];
+  var parent = getParent(element, 'bpmn:Process');
+  if (parent) {
+    return true;
+  }
 
-  var parent = getParent(element, types);
+  parent = getParent(element, 'bpmn:Participant');
   if (parent) {
     return isHorizontal(parent);
-  } else if (isAny(element, types)) {
+  } else if (is(element, 'bpmn:Participant')) {
     return isHorizontal(element);
   }
 
-  return true;
+  var process;
+  for (process = getBusinessObject(element); process; process = process.$parent) {
+    if (is(process, 'bpmn:Process')) {
+      break;
+    }
+  }
+
+  // The direction may be specified in another diagram. We ignore that there
+  // could be multiple diagrams with contradicting properties based on the
+  // assumption that such BPMN files are unusual.
+  var pool = elementRegistry.find(function(shape) {
+    var businessObject = getBusinessObject(shape);
+    return businessObject && businessObject.get('processRef') === process;
+  });
```

Comment:
Unfortunately, for some elements on a collaboration diagram `process` ends up being undefined, and this leads to unexpected outcomes.

https://github.com/user-attachments/assets/b95c153f-a106-4243-a73d-6fd33897dd93